datastream-server/src/test/java/com/linkedin/datastream/server/zk/TestZkAdapter.java

```diff
@@ -759,6 +759,26 @@ public void testDeleteTasksWithPrefix() {
     List<String> leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
     Assert.assertEquals(leftOverTasks.size(), 2);
 
+    adapter.cleanUpOrphanConnectorTasks(false);
+
+    leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
+    Assert.assertEquals(leftOverTasks.size(), 2);
+
+    adapter.cleanUpOrphanConnectorTasks(true);
+
+    leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
+    Assert.assertEquals(leftOverTasks.size(), 2);
+
+    updateInstanceAssignment(adapter, adapter.getInstanceName(), new ArrayList<DatastreamTask>());
+
+    leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
+    Assert.assertEquals(leftOverTasks.size(), 2);
+
+    adapter.cleanUpOrphanConnectorTasks(true);
```

Comment:
should we test with false first after updateInstanceAssignments and see 2 left as well?

Reply:
Done.